overlay/master.cpp

```diff
@@ -761,24 +963,53 @@ class ManagerProcess : public ProtobufProcess<ManagerProcess>
           "Unable to parse VTEP MAC OUI: " + vtepMACOUI.error());
     }
 
+    // IPv6
+    Option<Network> vtepSubnet6 = None();
+    if (networkConfig.has_vtep_subnet6()) {
+      Try<Network> _vtepSubnet6 =
+        Network::parse(networkConfig.vtep_subnet6(), AF_INET6);
+      if (_vtepSubnet6.isError()) {
+        return Error(
+            "Unable to parse the VTEP IPv6 Subnet: " + _vtepSubnet6.error());
+      }
+      vtepSubnet6 = _vtepSubnet6.get();
+    }
+
     hashmap<string, Owned<Overlay>> overlays;
-    IntervalSet<uint32_t> addressSpace;
+    IntervalSet<IP> addressSpace;
 
     // Overlay networks cannot have overlapping IP addresses. This
     // lambda keeps track of the current address space and returns an
     // `Error` if it detects an overlay that is going to use an
     // already configured address space.
     auto updateAddressSpace =
-      [&addressSpace](const net::IP::Network &network) -> Try<Nothing> {
-        uint32_t startIP = ntohl(network.address().in().get().s_addr);
-
-        uint32_t mask = ntohl(network.netmask().in().get().s_addr);
-        mask = ~mask;
-
-        uint32_t endIP = startIP | mask;
+      [&addressSpace](const Network &network) -> Try<Nothing> {
+        IP startIP, endIP;
+        switch(network.address().family()) {
+          case AF_INET: {
```

Comment:
Feel like `startIP` and `endIP` should be `start` and `end` methods in `overlay::Network`. Would really cleanup this code here.

Reply:
Looks like we already have implementation of `begin` and `end` in `overlay::Network`. So we should be able to use it over here?